src/NHibernate.Config.Templates/FireBird.cfg.xml

```diff
@@ -22,6 +22,7 @@ for your own use before compile tests in VisualStudio.
 			Database=nhibernate;
 			User ID=SYSDBA;Password=masterkey;
 			MaxPoolSize=200;
+			charset=utf8;
```

Comment:
Default charset for FireBird is "None", theoretically meaning that it is the application business to encode/decode it, Firebird being supposed to stores and gives data back "as is", maybe in some binary sens. In our case, that goes wrong. I have not investigated why, I have just switched that to utf8 instead.

This requires to recreate the database, which the TeamCity build does for Firebird. For local tests, adjust your configuration files accordingly and run the TestDatabaseSetup test for fixing this.

(Of course this file here is just the default template, the actual fix for TeamCity is the similar change done in teamcity.build file.)